src/main/java/sg/ncl/DataController.java

```diff
@@ -203,6 +204,124 @@ public String removeDataset(@PathVariable String id, @PathVariable Optional<Stri
+                } else if (exceptionState == DATA_NOT_FOUND_EXCEPTION) {
+                    log.warn("Dataset not found for requesting access.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else {
+                    log.warn("Unknown error for requesting dataset access.");
+                }
+            } else {
+                log.info("Dataset access requested: {}", dataResponseBody);
+            }
+        } catch (IOException e) {
+            log.error("requestDataset: {}", e.toString());
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return REDIRECT_DATA;
+    }
+
+    @RequestMapping(value = "{did}/requests/{rid}", method = RequestMethod.GET)
+    public String getRequest(Model model, @PathVariable String did, @PathVariable String rid) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getRequest(did, rid), HttpMethod.GET, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == FORBIDDEN_EXCEPTION) {
+                    log.warn("Requesting of dataset access forbidden.");
+                    model.addAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_ACCESS_REQUEST_NOT_FOUND_EXCEPTION) {
+                    log.warn("Dataset access request not found.");
+                    model.addAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_NOT_MATCH_EXCEPTION) {
+                    log.warn("Dataset access request does not have matching parent.");
+                    model.addAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else if (exceptionState == DATA_NOT_FOUND_EXCEPTION) {
+                    log.warn("Dataset not found for requesting access.");
+                    model.addAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                } else {
+                    log.warn("Unknown error for getting dataset access request.");
+                }
+            } else {
+                log.info("Dataset access request retrieved: {}", dataResponseBody);
+                DataAccessRequest dataAccessRequest = extractDataAccessRequest(dataResponseBody);
+                model.addAttribute("dataAccessRequest", dataAccessRequest);
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)